gloo/docs/user_guides/tls_setup.md

```diff
@@ -0,0 +1,317 @@
+---
+title: Setting up Server TLS
+weight: 50
+description: Understanding how to set up TLS for Gloo
+---
+
+## Configuring TLS for Gloo
+
+We can configure HTTPS for our Gloo services to encrypt traffic coming from external clients. We can also configure Gloo to do mTLS with external clients as well. In this document, we'll explore configuring Gloo for server TLS.
+
+## Server TLS
+Gloo supports server-side TLS where the server presents a certificate to the client based on the domain for what the client asked. This means we can support multiple virtual hosts on a single port and use SNI to determine what certificate to serve depending what domain the client is requesting. In Gloo, we associate our TLS configuration with a specific VirtualService which can then describe which SNI hosts would be candidates for both the TLS certificates as well as the routing rules that are defined in the VirtualService. Let's look at setting up TLS.
```

Comment:
Since you're covering 2 related use cases in this doc - setting up one TLS end point going to an upstream service; and setting up 2 TLS end points going to an upstream - you may want to start that more clearly. That is, that doc will cover 2 related use cases. Reading through the doc, I didn't initially realized I'd started reading a 2nd use case, and wondered why you started talking about creating a 2nd cert...

Reply:
Interesting. Okay I'm open to try make this clear. So first thing to make clear, we're not setting up TLS to upstreams in this doc. I almost added that, but I think it would probably be best in another doc. Second thing... this is really just one use case: setting up VirtualHosts with TLS... but the implicit expectation here is that there can be more than one VirtualHost. Let me go back through and see if I can make that clear.